apps/compound/src/App.tsx

```diff
@@ -1,57 +1,34 @@
 import Big from 'big.js';
-import { BigNumberInput } from 'big-number-input';
-import Web3 from 'web3';
-import { AbiItem } from 'web3-utils';
-
-import { Button, Select, Title, Section, Text, TextField, Divider, Loader } from '@gnosis.pm/safe-react-components';
+import { Button, Select, Title, Section, Text, Loader } from '@gnosis.pm/safe-react-components';
 import { useSafeAppsSDK } from '@gnosis.pm/safe-apps-react-sdk';
-import styled from 'styled-components';
-
-import { rpc_token, getTokenList, TokenItem } from './config';
-import { WidgetWrapper, SelectContainer, DaiInfo, ButtonContainer } from './components';
+import { getTokenList, TokenItem } from './config';
+import { SelectContainer, InfoContainer, ButtonContainer, StyledTitle, StyledTextField } from './styles';
 import { getTokenInteractions, parseEvents } from './tokensTransfers';
-import { networkByChainId, CHAINS } from './utils/networks';
-
-import cERC20Abi from './abis/CErc20';
-import cWEthAbi from './abis/CWEth';
-
-const blocksPerDay = 5760;
+import useComp from './hooks/useComp';
+import useWeb3 from './hooks/useWeb3';
+import useCToken from './hooks/useCToken';
+import CompBalance from './components/CompBalance';
+import InfoRow from './components/InfoRow';
+import WidgetWrapper from './components/WidgetWrapper';
+import { BigNumberInput } from 'big-number-input';
 
 type Operation = 'lock' | 'withdraw';
 
-const StyledTitle = styled(Title)`
-  margin-top: 0;
-`;
-
 const CompoundWidget = () => {
-  const [web3, setWeb3] = useState<Web3 | undefined>();
-  const { sdk: appsSdk, safe: safeInfo, connected } = useSafeAppsSDK();
   const [ethBalance, setEthBalance] = useState('0');
-
   const [tokenList, setTokenList] = useState<Array<TokenItem>>();
-
   const [selectedToken, setSelectedToken] = useState<TokenItem>();
-  const [cTokenInstance, setCTokenInstance] = useState<any>();
-  const [tokenInstance, setTokenInstance] = useState<any>();
-
-  const [cTokenSupplyAPY, setCTokenSupplyAPY] = useState('0');
+  const { cTokenInstance, tokenInstance } = useCToken(selectedToken);
   const [interestEarn, setInterestEarn] = useState('0');
   const [tokenBalance, setTokenBalance] = useState<string>('0');
   const [underlyingBalance, setUnderlyingBalance] = useState<string>('0');
-
   const [inputValue, setInputValue] = useState<string>('');
   const [inputError, setInputError] = useState<string | undefined>();
-
-  // set web3 instance
-  useEffect(() => {
-    if (!safeInfo) {
-      return;
-    }
-    const chainId = safeInfo.chainId as CHAINS;
-    const web3Instance = new Web3(`https://${networkByChainId[chainId]}.infura.io/v3/${rpc_token}`);
-    setWeb3(web3Instance);
-  }, [safeInfo]);
+  const { web3 } = useWeb3();
+  const { sdk: appsSdk, safe: safeInfo, connected } = useSafeAppsSDK();
+  const { cTokenSupplyAPY, cDistributionTokenSupplyAPY, claimableComp, claimComp } = useComp(selectedToken);
+  const isMainnet = useRef(false);
```

Comment:
I think `useMemo` suits better here

Reply:
mmm why do you think so. I can change ir but not sure about the benefit. useMemo cause rerender

Reply:
So, what you're doing now:
```
{isMainnet.current && (
            <InfoRow label="Distribution APY" data={cDistributionTokenSupplyAPY && `${cDistributionTokenSupplyAPY}%`} />
          )}
```

React doesn't track changes for `isMainnet.current` (that's why `useRef` is needed)
So, if isMainnet.current changes, the react won't update the output markup. And the reason why it works is because the app re-rendered after the value was set

Reply:
mmm right. I was thinking it as a load and forget variable because if user change network in safe web the ui is really reloading the web application and we need to enter the app again. But you are right let me change it for being future proof

Reply:
done, you can review it